packages/astro/src/content/utils.ts

```diff
@@ -358,3 +387,90 @@ function search(fs: typeof fsMod, srcDir: URL) {
 	}
 	return { exists: false, url: paths[0] };
 }
+
+export async function getStringifiedLookupMap({
+	contentPaths,
+	contentEntryConfigByExt,
+	root,
+	fs,
+}: {
+	contentEntryConfigByExt: ReturnType<typeof getContentEntryConfigByExtMap>;
+	contentPaths: Pick<ContentPaths, 'contentDir' | 'cacheDir'>;
+	root: URL;
+	fs: typeof fsMod;
+}) {
+	const { contentDir } = contentPaths;
+	const globOpts: FastGlobOptions = {
+		absolute: true,
+		cwd: fileURLToPath(root),
+		fs: {
+			readdir: fs.readdir.bind(fs),
+			readdirSync: fs.readdirSync.bind(fs),
+		},
+	};
+
+	const relContentDir = rootRelativePath(root, contentDir, false);
+	const contentGlob = await glob(
+		`${relContentDir}**/*${getExtGlob([...contentEntryConfigByExt.keys()])}`,
+		globOpts
+	);
+	let filePathByLookupId: {
+		[collection: string]: Record<string, string>;
+	} = {};
+
+	await Promise.all(
+		contentGlob.map(async (filePath) => {
+			const info = getEntryInfo({ contentDir, entry: filePath });
+			if (info instanceof NoCollectionError) return;
+			const contentEntryType = contentEntryConfigByExt.get(extname(filePath));
+			if (!contentEntryType) return;
+
+			const { id, collection, slug: generatedSlug } = info;
+			filePathByLookupId[collection] ??= {};
+			const slug = await getEntrySlug({
+				id,
+				collection,
+				generatedSlug,
+				fs,
+				fileUrl: pathToFileURL(filePath),
+				contentEntryType,
+			});
+			filePathByLookupId[collection][slug] = rootRelativePath(root, filePath);
```

Comment:
Since `filePathByLookupId[collection]` is going to have a `[slug]` either way, should this work too?

```js
const slug = await getEntrySlug({
	id,
	collection,
	generatedSlug,
	fs,
	fileUrl: pathToFileURL(filePath),
	contentEntryType,
});
filePathByLookupId[collection] = {
	...filePathByLookupId[collection],
	[slug]: rootRelativePath(root, filePath)
}
```

If so, I like more this approach because it's easier to read compared to the previous (the `??=` was unexpected XD)

Reply:
Ugh every time I use `??=` people complain 😅  Yeah, this should work too but it's 50% less fun